### PR TITLE
Describe filtered responses better

### LINKED ISSUE #1509: Describe filtered responses better
There's a couple things that need to be done:

1. There's a number of call sites that assume filtered responses have the same associated concepts as normal responses do. They probably should, except if this causes a problem of some kind. Needs to be double checked and then we should update the definition of a filtered response to make it clear that unless stated otherwise it has all the same associated concepts.
2. It would be good to check if response call sites in general use the filtered response correctly as I haven't been as disciplined about this as I should have been.

### PR DESCRIPTION
**Pull request by annevk**
Fixes #1509. Follow-ups: #1511 & #1512.

***
<a href="https://whatpr.org/fetch/1513.html" title="Last updated on Oct 25, 2022, 8:44 AM UTC (9f5e5c4)">Preview</a> | <a href="https://whatpr.org/fetch/1513/6291752...9f5e5c4.html" title="Last updated on Oct 25, 2022, 8:44 AM UTC (9f5e5c4)">Diff</a>

### CODE REVIEW

File: fetch.bs [hunk 1]
```diff
@@ -2304,18 +2304,28 @@ known as an <dfn export id=concept-aborted-network-error>aborted network error</
...
+<a for=/>response</a> that is neither a <a for=/>network error</a> nor a
+<a for=/>filtered response</a>).
+
+<p>Unless stated otherwise a <a for=/>filtered response</a>'s associated concepts (such as its
+<a for=response>body</a>) refer to the associated concepts of its
+<a for="filtered response">internal response</a>. (The exceptions to this are listed below as part
+of defining the concrete types of <a for=/>filtered responses</a>.)
+
+<div class=note>
+ <p>The <a for=/>fetch</a> algorithm returns such a view to ensure APIs do not accidentally leak
```
Comment (domenic) [inline fetch.bs:2320]: Fetch no longer "returns" responses, so maybe tweak this?

File: fetch.bs [hunk 2]
```diff
@@ -2304,18 +2304,28 @@ known as an <dfn export id=concept-aborted-network-error>aborted network error</
...
+ information. If the information needs to be exposed for legacy reasons, e.g., to feed image data to
+ a decoder, the associated <a for="filtered response">internal response</a> can be used, which is
+ only "accessible" to internal specification algorithms.
```
Comment (domenic) [inline fetch.bs:2323]: What internal specification algorithms is it accessible to?

File: fetch.bs [hunk 3]
```diff
@@ -2304,18 +2304,28 @@ known as an <dfn export id=concept-aborted-network-error>aborted network error</
...
+
+ <p>New specifications ought not to build further on <a>opaque filtered responses</a> or
+ <a>opaque-redirect filtered responses</a>. Those are legacy constructs and cannot always be
+ adequately protected given contemporary computer architecture.
+</div>
 
 <p>A <dfn export id=concept-filtered-response-basic>basic filtered response</dfn> is a
```
Comment (domenic) [inline fetch.bs:2330]: Maybe out of scope / better as a followup, but I don't know what each of these types are used for. Especially "basic filtered response", and why there are two types of responses that sound similar ("basic" vs. "default"). Examples would be helpful.

File: fetch.bs (hunk 2 above)
Comment (annevk) [reply fetch.bs:2323]: Any that need access? E.g., scripts, images, style sheets.

File: fetch.bs (hunk 3 above)
Comment (annevk) [reply fetch.bs:2330]: "basic" is essentially about hiding cookies. "default" is a response that's not filtered at all. I think you only get that with synthetic responses that haven't gone through fetch.

File: fetch.bs (hunk 2 above)
Comment (domenic) [reply fetch.bs]: I stil find this sentence confusing them... what is an example of a non-internal specification algorithm, which doesn't have access?

File: fetch.bs [hunk 4]
```diff
@@ -2371,6 +2382,32 @@ is a <a>filtered response</a> whose
  <a attribute for=Response lt=ok><code>response.ok</code></a>, will return rather useless results.
 </div>
 
+<div class=example id=example-filtered-responses>
+ <p>The <a for=response>type</a> of a <a for=/>response</a> is exposed to script through the
+ {{Response/type}} getter:
+
+ <pre><code class=lang-javascript>
+(async function filteredResponseDemo() {
```
Comment (domenic) [inline fetch.bs:2390]: You can assume top-level await now, IMO, so the wrapper is no longer necessary :).

File: fetch.bs [hunk 5]
```diff
@@ -2304,18 +2304,29 @@ known as an <dfn export id=concept-aborted-network-error>aborted network error</
...
-<dfn export id=concept-internal-response for="filtered response">internal response</dfn>.
-
-<p class="note no-backref">The <a for=/>fetch</a> algorithm returns such a view to ensure APIs do
-not accidentally leak information. If the information needs to be exposed for legacy reasons, e.g.,
-to feed image data to a decoder, the associated <a for="filtered response">internal response</a> can
-be used, which is only "accessible" to internal specification algorithms and is never a
-<a>filtered response</a> itself.
+<p>A <dfn export id=concept-filtered-response>filtered response</dfn> is a <a for=/>response</a>
+that offers a limited view on an associated <a for=/>response</a>. This associated
+<a for=/>response</a> can be accessed through <a>filtered response</a>'s associated
```
Comment (domenic) [inline fetch.bs:2309]:
```suggestion
<a for=/>response</a> can be accessed through <a>filtered response</a>'s
```

to reduce the repetition of "associated"

File: fetch.bs (hunk 2 above)
Comment (annevk) [reply fetch.bs]: Oh, I could remove the word "internal".

**Review by domenic — APPROVED** [review]

File: fetch.bs [hunk 6]
```diff
@@ -2304,18 +2304,29 @@ known as an <dfn export id=concept-aborted-network-error>aborted network error</
...
+<a for=response>body</a>) refer to the associated concepts of its
+<a for="filtered response">internal response</a>. (The exceptions to this are listed below as part
+of defining the concrete types of <a for=/>filtered responses</a>.)
+
+<div class=note>
+ <p>The <a for=/>fetch</a> algorithm by way of <a for=fetch><i>processResponse</i></a> and
+ equivalent parameters exposes <a for=/>filtered responses</a> to callers to ensure they do not
+ accidentally leak information. If the information needs to be revealed for legacy reasons, e.g., to
+ feed image data to a decoder, the associated <a for="filtered response">internal response</a> can
+ only be used by specification algorithms.
```
Comment (annevk) [inline fetch.bs:2324]:
```suggestion
 be used by specification algorithms.
```

**Pull request merged**